node/impl/client/client.go

```diff
@@ -196,17 +197,31 @@ func (a *API) ClientFindData(ctx context.Context, root cid.Cid) ([]api.QueryOffe
 	return out, nil
 }
 
-func (a *API) ClientImport(ctx context.Context, path string) (cid.Cid, error) {
-	f, err := os.Open(path)
+func (a *API) ClientImport(ctx context.Context, ref api.FileRef) (cid.Cid, error) {
+	f, err := os.Open(ref.Path)
 	if err != nil {
 		return cid.Undef, err
 	}
+
+	if ref.IsCAR {
+		result, err := car.LoadCar(a.Blockstore, f)
```

Comment:
Just a note, but this will meant that importing a car file doesnt use the filestore

Reply:
yea I saw that.